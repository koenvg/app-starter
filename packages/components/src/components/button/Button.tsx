import React, { SFC } from "react";
import { Button as MUIButton, ButtonProps, CircularProgress } from "@material-ui/core";

export interface Props extends ButtonProps {
  loading?: boolean;
}

export const Button: SFC<Props> = ({ children, loading, ...rest }) => {
  return (
    <MUIButton {...rest} disabled={loading || rest.disabled} disableRipple>
      {loading && <CircularProgress size={25} />}
      {!loading && children}
    </MUIButton>
  );
};

Button.displayName = "Button";
