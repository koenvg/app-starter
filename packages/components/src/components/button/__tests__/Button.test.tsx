import { Button, Props } from "../Button";
import React from "react";
import { render } from "@testing-library/react";

describe("Button", () => {
  const defaultProps: Props = {};

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render the button", () => {
    const component = render(<Button {...defaultProps}>Click me</Button>);
    expect(component.container).toMatchSnapshot();
  });

  it("should render a loading indicator and disable the button when loading is true", () => {
    const component = render(
      <Button {...defaultProps} loading>
        Click me
      </Button>,
    );
    expect(component.container).toMatchSnapshot();
  });
});
