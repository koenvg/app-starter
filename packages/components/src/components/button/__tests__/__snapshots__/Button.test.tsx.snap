// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a loading indicator and disable the button when loading is true 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
    disabled=""
    tabindex="-1"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 25px; height: 25px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </span>
  </button>
</div>
`;

exports[`Button should render the button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Click me
    </span>
  </button>
</div>
`;
